src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoSurveyDaoTest.java

```diff
@@ -675,6 +676,92 @@ public void constraintRulesOnQuestionOverwriteConstraintRules() {
         assertTrue(rule.getEndSurvey());
     }
     
+    @Test
+    public void getSurveyAllVersionsIncludeDeletedByIdentifier() {
+        List<DynamoSurvey> surveyList = ImmutableList.of(new DynamoSurvey(), new DynamoSurvey());
+        mockSurveyMapper(surveyList);
+        
+        
+        List<Survey> results = dao.getSurveyAllVersions(TEST_STUDY, SURVEY_IDENTIFIER, true);
+        assertEquals(results, surveyList);
+        
+        verify(mockSurveyMapper).queryPage(eq(DynamoSurvey.class), queryCaptor.capture());
+        DynamoDBQueryExpression<DynamoSurvey> query = queryCaptor.getValue();
+        assertEquals(query.getHashKeyValues().getIdentifier(), SURVEY_IDENTIFIER.substring(11));
```

Comment:
It's more important to test the studyIdentifier, since that's the actual hash key. (Here and in all the other tests.)